rotkehlchen/db/upgrades/v41_v42.py

```diff
@@ -198,6 +198,21 @@ def _remove_balancer_events_table(write_cursor: 'DBCursor') -> None:
     )
 
 
+def _delete_orphan_events(write_cursor: 'DBCursor') -> None:
+    """
+    Delete all the evm events that have a tx_hash that is not present in the db. This can happen
+    for customized events of addresses that got deleted from the database for example.
+
+    This query assumes that an evm event always has a evm_events_info but this is not the case
+    for zksync lite. Since prior to 1.33 that assumption holds this query is safe.
+    """
+    write_cursor.execute(
+        'DELETE FROM history_events WHERE identifier IN (SELECT identifier FROM evm_events_info '
+        'WHERE identifier NOT IN (SELECT eei.identifier FROM evm_events_info eei JOIN '
```

Comment:
Isn't this simpler?

```suggestion
        'DELETE FROM history_events WHERE identifier NOT IN (SELECT eei.identifier FROM evm_events_info eei JOIN '
```

Reply:
Is not the same query since yours also deletes kraken events for example